api/src/main/java/com/codeforcommunity/rest/subrouter/TeamsRouter.java

```diff
@@ -22,12 +28,137 @@ public TeamsRouter(ITeamsProcessor processor) {
+    registerKickUserRoute(router);
+    registerLeaveTeamRoute(router);
+    registerTransferTeamOwnership(router);
 
     return router;
   }
 
+  // Create a Team
+  private void registerCreateTeam(Router router) {
+    Route createTeamRoute = router.post("/create");
+    createTeamRoute.handler(this::handleCreateTeamRoute);
+  }
+
+  private void handleCreateTeamRoute(RoutingContext routingContext) {
+    JWTData userData = routingContext.get("jwt_data");
+    CreateTeamRequest createTeamRequest =
+        RestFunctions.getJsonBodyAsClass(routingContext, CreateTeamRequest.class);
+    processor.createTeam(userData, createTeamRequest);
+    end(routingContext.response(), 200);
+  }
+
+  // Get a Team
+  private void registerGetTeam(Router router) {
+    Route getTeamRoute = router.get("/:team_id");
+    getTeamRoute.handler(this::handleGetTeamRoute);
+  }
+
+  private void handleGetTeamRoute(RoutingContext routingContext) {
+    JWTData userData = routingContext.get("jwt_data");
+    int teamId = RestFunctions.getRequestParameterAsInt(routingContext.request(), "team_id");
+    TeamDataResponse getTeamResponse = processor.getTeam(userData, teamId);
+    end(routingContext.response(), 200, JsonObject.mapFrom(getTeamResponse).toString());
+  }
+
+  // Add a Goal
+  private void registerAddGoal(Router router) {
+    Route addGoalRoute = router.post("/:team_id/add_goal");
+    addGoalRoute.handler(this::handleAddGoal);
+  }
+
+  private void handleAddGoal(RoutingContext routingContext) {
+    JWTData userData = routingContext.get("jwt_data");
+    int teamId = RestFunctions.getRequestParameterAsInt(routingContext.request(), "team_id");
+    AddGoalRequest addGoalRequest =
+        RestFunctions.getJsonBodyAsClass(routingContext, AddGoalRequest.class);
+    processor.addGoal(userData, addGoalRequest, teamId);
+    end(routingContext.response(), 200);
+  }
+
+  // Delete a Goal
+  private void registerDeleteGoal(Router router) {
+    Route deleteGoalRoute = router.post("/:team_id/delete_goal/:goal_id");
+    deleteGoalRoute.handler(this::handleDeleteGoal);
+  }
+
+  private void handleDeleteGoal(RoutingContext routingContext) {
+    JWTData userData = routingContext.get("jwt_data");
+    int teamId = RestFunctions.getRequestParameterAsInt(routingContext.request(), "team_id");
+    int goalId = RestFunctions.getRequestParameterAsInt(routingContext.request(), "goal_id");
+    processor.deleteGoal(userData, teamId, goalId);
+    end(routingContext.response(), 200);
+  }
+
+  private void registerInviteUser(Router router) {
+    Route inviteUserRoute = router.post("/:team_id/invite");
+    inviteUserRoute.handler(this::handleInviteUser);
+  }
+
+  private void handleInviteUser(RoutingContext routingContext) {
+    JWTData userData = routingContext.get("jwt_data");
+    InviteUserRequest inviteUserRequest =
+        RestFunctions.getJsonBodyAsClass(routingContext, InviteUserRequest.class);
+    processor.inviteUser(userData, inviteUserRequest);
```

Comment:
No response

Reply:
oops forgot to push a final commit here...